src/emulator/storage/apis/firebase.ts

```diff
@@ -128,6 +130,48 @@ export function createFirebaseEndpoints(emulator: StorageEmulator): Router {
     return res.json(new OutgoingFirebaseMetadata(metadata));
   });
 
+  firebaseStorageAPI.post(`/b/:bucketId/o/:objectId[(:)]generateSignedUrl`, async (req, res) => {
+    const timeToLive = req.body.ttl;
+    const msInAWeek = 7 * 24 * 60 * 60 * 1000;
+
+    if (timeToLive < 0 || timeToLive > msInAWeek) {
+      return res.status(400).json({
+        error: {
+          code: 400,
+          message: `Invalid TTL.`,
+        },
+      });
+    }
+    const currentDate = new Date()
+      .toISOString()
+      .replaceAll("-", "")
+      .replaceAll(":", "")
+      .replaceAll(".", "");
+
+    const unsignedUrl = `${req.protocol}://${req.hostname}:${req.socket.localPort}/v0/b/${req.params.bucketId}/o/${req.params.objectId}?alt=media&X-Firebase-Date=${currentDate}&X-Firebase-Expires=${timeToLive}`;
+
+    const authorized = await storageLayer.authenticateUser({
```

Comment:
Could you move the security rules check to the top of the route. If they fail, we should exit with 403 immediately.

Reply:
naming nit: `authenticate` is different than `authorize`: https://auth0.com/docs/get-started/identity-fundamentals/authentication-and-authorization

Reply:
Discussed offline, rulesValidator is a private member of storageLayer, so the new `generateSignedUrl` should call it.